backend_plugin/src/test/java/negatives/evaluators/AddImportsNegative.java

```diff
@@ -0,0 +1,335 @@
+package test.java.negatives.evaluators;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import org.eclipse.jface.text.BadLocationException;
+import org.eclipse.jface.text.Document;
+import org.eclipse.jface.text.DocumentEvent;
+import org.eclipse.jface.text.IDocument;
+import org.eclipse.jface.text.Position;
+import org.eclipse.jface.text.source.Annotation;
+import org.eclipse.jface.text.source.AnnotationModel;
+import org.junit.Before;
+import org.junit.Test;
+
+import main.evaluators.AddImportEvaluator;
+
+public class AddImportsNegative {
+	
+	/**
+	 * Constants used as mock data. INITIAL_CONTENT is the content of the document,
+	 * ANNOTATION represents an unresolved type annotation, and POSITION is a variable
+	 * necessary for adding the annotation to the annotation model
+	 */
+	private static final String INITIAL_CONTENT = "\n\nLine1\n Line2\n Line3\n";
+	private static final Annotation ANNOTATION = new Annotation("org.eclipse.jdt.ui.error", 
+			false, "MockType cannot be resolved to a type");
+	private static final Position POSITION = new Position(0);
+	private static final String MOCK_IMPORT = "import java.util.*;";
+	
+	private AddImportEvaluator eval;
+	private IDocument doc;
+	private AnnotationModel am;
+
+	/**
+	 * Initialize a new document and a new evaluator
+	 */
+	@Before
+	public void runBeforeTests() {
+		this.doc = new Document(INITIAL_CONTENT);
+		this.am = new AnnotationModel();
+		this.eval = new AddImportEvaluator(doc);
+	}
+
+	/**
+	 * Tests the user starting a valid import statement by typing
+	 * "import " one character at a time when there are no unresolved
+	 * variables
+	 */
+	@Test
+	public void noUnresolvedTypesAddImportOneCharacterAtATime() {
+		try {
+			int offset = 0;
+			assertFalse(mockUserInput("i", offset++));
```

Comment:
To make this more readable, you can use `for (char c : "import".toCharArray())`

Reply:
Good call, I've refactored the class to simplify these actions as noted